lib/orderbook/TradingPair.ts

```diff
@@ -299,41 +295,43 @@ class TradingPair {
 
     // as long as we have remaining quantity to match and orders to match against, keep checking for matches
     while (remainingOrder && !queue.isEmpty()) {
```

Comment:
That'd be fine with me.

Reply:
This turned out to be a bit more complicated than I anticipated since we're reassigning values to `remainingOrder`. It's still a good idea though, I'll try to deal with it in a future PR.

Reply:
No worries.